### PR TITLE
Upgrade from NUnit 2.6.4 to NUnit 3.10 and Visual Studio 2017.

### PR DESCRIPTION
**Pull request by monty241**
When rebuilding the assembly, the dependencies have bene updated. This pull requests changes the code and NuGet dependencies such that solution works with VS2017 and NUnit 3.10.

NUnit 3 has no longer support for Throws, so code changes are based upon https://github.com/nunit/nunit/issues/1107

### CODE REVIEW

**Comment by nathanpjones** [comment]
Thanks for the PR--updating for .NET Standard is something I've wanted to do for a while. Haven't had a chance to review it. Does it still maintain a PCL build to avoid breaking backwards compatibility?

**Comment by monty241** [comment]
Hi @nathanpjones , you are welcome. I doubt that it is compatible with PCL build, or let me rephrase: I wouldn't recognize a PCL build when I would come across it on the street; last exposure is long time ago. We solely use 4.7 and S2.0. Multi-platform builds in VS 2017 are still buggy in combination with our code base, so in general we create two projects and two solutions when a separate 4.7 and Standard 2.0 release are needed.

**Comment by monty241** [comment]
Update is that multi-platform builds in 15.9.6 of Visual Studio seem reliable now.

**Comment by monty241** [comment]
Did some catching up with reading; PCL is platform dependent set of libraries, .net standard is platform independent version of it.  Updated csproj checked in which build for both 4.7.2 and netstandard 2. More can be reliably added using recent versions of VS 2017.

**Comment by cdonnellytx** [comment]
Is there more that needs to be done for this? Can I help in any way?

**Comment by nathanpjones** [comment]
@monty241 and @cdonnellytx Sorry, been having some work life balance and health issues. I'm going to work on updating this and pushing the changes to NuGet.
